pipeline/matching/generators/similar_intents.py

```diff
@@ -15,4 +25,65 @@ def __init__(self, min_common: int = 0):
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
         # TODO: Implement your generator
-        yield from []
+        # inp.logger.info("Lauren")
+
+        users = inp.users
+        # print(users)
+        match = False
+
+        match_list = []
+
+        # iterate over the arr of users
+        for idx, u in enumerate(users):
+            # take Intent info from the current user
+            intents = u.intents
+            intent = intents[0]
+            code = intent.code
+            side = intent.side
+            name = intent.name
+
+            # o therSide would be the other side of the intent that they have
+            if side == Side.GIVING:
+                otherSide = Side.SEEKING
+            elif side == Side.SEEKING:
+                otherSide = Side.GIVING
+            else:
+                otherSide = Side.BLANK
+
+            # create an Intent object to search for in the other users
+            otherIntent = Intent(code, otherSide, name)
+
+            for oth in users:
+                if otherIntent in oth.intents:
+                    # create the intent match after finding a user that matches
+                    intent_match = IntentMatch(
+                        code,
+                        seeker=oth.uid
+                        if otherIntent.side == Side.SEEKING
+                        else u.uid,
+                        giver=oth.uid
+                        if otherIntent.side == Side.GIVING
+                        else u.uid,
+                    )
+
+                    # create a Match
+                    new_match = Match(
+                        users={u.uid, oth.uid},
+                        metadata=MatchMetadata(
+                            generator=GENERATOR_SIMILAR_INTENTS,
+                            score=1,
```

Comment:
You can exclude score and then it will get 0 as its default value. You can check if a dataclass has default values by looking at the definition, for example: https://github.com/scarletstudio/butterfly/blob/20edebb83c443a7358709f4966190e7b5e0aa3ca/pipeline/types/match_metadata.py#L35

Reply:
Good question, the ranker can add the score, or the generator can provide it. I have a slight preference for the ranker adding the score, that way the ranker controls scoring and thus ranking.

We can leave the score blank for now, this could be a new issue to add a score for these matches.